File: examples/integration/scheduler-client.py
# examples/integration/scheduler-client.py
# Python Example Client for CON-COST Dev Scheduler Integration API v1

import os
import requests

BASE_URL = os.getenv('SCHEDULER_API_URL', 'https://concost-dev-scheduler-qa.eumditravel.workers.dev')
API_KEY = os.getenv('SCHEDULER_API_KEY', 'sched_live_example_token')

headers = {
    'Content-Type': 'application/json',
    'Authorization': f'Bearer {API_KEY}'
}

# 1. Health Check
resp = requests.get(f'{BASE_URL}/api/integrations/v1/health')
print('Health:', resp.json())

# 2. Upsert Task
payload = {
    'source': 'python-script',
    'external_id': 'py-task-101',
    'project': {
        'external_id': 'concost-hub',
        'name': 'CONCOST-HUB Development'
    },
    'task_name': 'Automated Data Pipeline Task',
    'start_date': '2026-08-15',
    'end_date': '2026-08-18',
    'assignees': [
        {'worker_id': 'wrk_03', 'allocation_percent': 50},
        {'worker_id': 'wrk_04', 'allocation_percent': 50}
    ]
}

upsert_resp = requests.post(f'{BASE_URL}/api/integrations/v1/tasks/upsert', json=payload, headers=headers)
print('Upsert Status:', upsert_resp.status_code)
print('Upsert Result:', upsert_resp.json())

File: examples/integration/scheduler-client.mjs
// examples/integration/scheduler-client.mjs
// Node.js ESM Example Client for CON-COST Dev Scheduler Integration API v1

const BASE_URL = process.env.SCHEDULER_API_URL || 'https://concost-dev-scheduler-qa.eumditravel.workers.dev';
const API_KEY = process.env.SCHEDULER_API_KEY || 'sched_live_example_token';

async function main() {
  // 1. Health Check
  const healthRes = await fetch(`${BASE_URL}/api/integrations/v1/health`);
  console.log('Health:', await healthRes.json());

  // 2. Fetch Active Workers
  const workersRes = await fetch(`${BASE_URL}/api/integrations/v1/workers`, {
    headers: { Authorization: `Bearer ${API_KEY}` },
  });
  console.log('Workers:', await workersRes.json());

  // 3. Upsert Task
  const upsertRes = await fetch(`${BASE_URL}/api/integrations/v1/tasks/upsert`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${API_KEY}`,
    },
    body: JSON.stringify({
      source: 'github',
      external_id: 'issue-789',
      project_external_id: 'concost-hub',
      task_name: 'Fix OAuth Callback Timeout',
      start_date: '2026-08-10',
      end_date: '2026-08-12',
      assignees: [{ worker_id: 'wrk_03', allocation_percent: 100 }],
    }),
  });

  console.log('Task Upsert Status:', upsertRes.status);
  console.log('Task Upsert Result:', await upsertRes.json());
}

main().catch(console.error);
